### PR TITLE
feat: Formal Verification attributes inside the SSA

### PR DESCRIPTION
**Pull request by Syndamia**
The code compiles, however it needs to be tested.

### CODE REVIEW

File: compiler/noirc_frontend/src/monomorphization/ast.rs [hunk 1]
```diff
@@ -277,7 +277,7 @@ pub struct Function {
     pub unconstrained: bool,
     pub inline_type: InlineType,
     pub func_sig: FunctionSignature,
-    pub formal_verification_attributes: Vec<ResolvedFvAttribute>,
+    pub formal_verification_expressions: Vec<Expression>,
```
Comment (Aristotelis2002) [inline compiler/noirc_frontend/src/monomorphization/ast.rs:280]: The way you have implemented it we are losing information if the `Expression` is for an ensures attribute or a require attribute. You should create an enum like the ResolvedFvAttribute which contains ensure/requires and their expressions

Comment (Aristotelis2002) [reply compiler/noirc_frontend/src/monomorphization/ast.rs:280]: Also clear on line 10 the unused imports.

**Review by Aristotelis2002 — CHANGES_REQUESTED** [review]

**Review by Aristotelis2002 — APPROVED** [review]
LGTM. Please merge after finishing the offline testing.

**Review by Aristotelis2002 — APPROVED** [review]
LGTM. Good documentation

**Review by Aristotelis2002 — APPROVED** [review]
LGTM.

**Comment by Syndamia** [comment]
To use this branch, you'll need to make some slight modifications, mainly turn off function inlining and force-enable formal verifications.

Inside `compiler/noirc_evaluator/src/ssa.rs` remove lines containing

```rust
.run_pass(Ssa::inline_functions[...], "After inlining:")
```

Inside `compiler/noirc_frontend/src/hir/mod.rs` change from `false` to `true` the `perform_formal_verification` values.

**Review by Aristotelis2002 — APPROVED** [review]
LGTM.

**Pull request merged**